src/main/java/org/opentripplanner/routing/trippattern/TripTimes.java

```diff
@@ -474,6 +478,66 @@ public boolean tripAcceptable(final State state0, final int stopIndex) {
         return true;
     }
 
+    /**
+     * Returns true if the current state chain contains a banned trip, or banned trip/stop index or banned trip sequence
+     */
+    public boolean tripOrTripSequenceIsBanned(State state0, final int stopIndex) {
+        final RoutingRequest options = state0.getOptions();
+
+        // check if an exact trip ID is banned
+        final BannedStopSet banned = options.bannedTrips.get(trip.getId());
```

Comment:
A better name for `banned` might be `bannedStopsForTrip`.

Reply:
Used Binh's suggestion in 6adec27faa2e744f94cd91e4ae4d337a5c25c7c7